testing-common/src/main/java/io/opentelemetry/instrumentation/testing/InstrumentationTestRunner.java

```diff
@@ -122,12 +122,19 @@ private <T extends Consumer<TraceAssert>> void waitAndAssertTraces(
     try {
       await()
           .untilAsserted(() -> doAssertTraces(traceComparator, assertionsList, verifyScopeVersion));
-    } catch (ConditionTimeoutException e) {
-      // Don't throw this failure since the stack is the awaitility thread, causing confusion.
-      // Instead, just assert one more time on the test thread, which will fail with a better stack
-      // trace.
-      // TODO(anuraaga): There is probably a better way to do this.
-      doAssertTraces(traceComparator, assertionsList, verifyScopeVersion);
+    } catch (Throwable t) {
+      // from org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:157)
+      // see https://github.com/oracle/graal/issues/6101 (spring boot graal native image)
+      if (t.getClass().getName().equals("com.oracle.svm.core.jdk.UnsupportedFeatureError")
```

Comment:
What is the JMX call leading to an issue? The MBeans provided by Spring Boot 3 are disabled by default.

Reply:
- This is the call: https://github.com/awaitility/awaitility/blob/fbe16add874b4260dd240108304d5c0be84eabc8/awaitility/src/main/java/org/awaitility/core/ConditionAwaiter.java#L157
- It only happens with native image
- It's not a bean provided by spring

Reply:
Could you please improve the comment?